CMakeLists.txt

```diff
@@ -0,0 +1,214 @@
+
+cmake_minimum_required(VERSION 3.4.3)
+
+list(APPEND CMAKE_MODULE_PATH
+     "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
+
+project(Foundation
+        LANGUAGES
+          C)
+
+option(FOUNDATION_PATH_TO_LIBDISPATCH_SOURCE "Path to libdispatch source" "")
+option(FOUNDATION_PATH_TO_LIBDISPATCH_BUILD "Path to libdispatch build" "")
+
+include(SwiftSupport)
+include(GNUInstallDirs)
+include(ExternalProject)
+
+ExternalProject_Add(CoreFoundation
+                    SOURCE_DIR
+                      ${CMAKE_CURRENT_SOURCE_DIR}/CoreFoundation
+                    CMAKE_COMMAND
+                      ${CMAKE_COMMAND}
+                    CMAKE_ARGS
+                      -DBUILD_SHARED_LIBS=NO
+                      -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+                      -DCMAKE_MAKE_PROGRAM=${CMAKE_MAKE_PROGRAM}
+                      -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+                      -DCMAKE_INSTALL_LIBDIR=usr/lib
+                      -DCF_DEPLOYMENT_SWIFT=YES
+                      -DCF_PATH_TO_LIBDISPATCH_SOURCE=${FOUNDATION_PATH_TO_LIBDISPATCH_SOURCE}
+                      -DCF_PATH_TO_LIBDISPATCH_BUILD=${FOUNDATION_PATH_TO_LIBDISPATCH_BUILD})
+ExternalProject_Get_Property(CoreFoundation install_dir)
+
+set(swift_optimization_flags)
+if(CMAKE_BUILD_TYPE MATCHES Release)
+  set(swift_optimization_flags -O)
+endif()
+add_swift_library(Foundation
```

Comment:
Does this build incrementally or as a giant glob?

Reply:
That is one of the areas of improvement :-).  It is a giant glob, but it should be possible to do it incrementally (I need to figure out how to do that).